File: practica2_820574_839304/java/Over.java
/*
* Fichero: Over.java
* Autores: Alicia Lázaro Huerta y Manel Jordá Puig Rubio
* Fecha: 27/02/2023
*/
import java.util.Stack;

public class Over extends Instruccion{
    Over(){
        nombre = "over";
    }
    public void ejecutar(Stack<Integer> pila, int[] pc){
        int a, b;
        a = pila.pop();
        b = pila.pop();

        pila.push(b);
        pila.push(a);
        pila.push(b);
        pc[0]++;
    }
}

File: practica2_820574_839304/c++/over.cc
/*
* Fichero: over.cc
* Autores: Alicia Lazaro Huerta y Manel Jorda Puig Rubio
* Fecha: 27/02/2023
*/

#include "over.h"

Over::Over() {
    nombre = "over";
}

void Over::ejecutar(stack<int> &pila, int &pc) {
    int a, b;
    a = pila.top();
    pila.pop();
    b = pila.top();
    pila.pop();

    pila.push(b);
    pila.push(a);
    pila.push(b);
    pc++;
}

string Over::nombrar() {
    return nombre;
}